golang/cmd/factoryinsight/database.go

```diff
@@ -1789,3 +1791,645 @@ func GetUniqueProductsWithTags(parentSpan opentracing.Span, customerID string, l
 
 	return
 }
+
+// GetAccumulatedProducts gets the accumulated counts for an observation timeframe and an asset
```

Comment:
Can you split it up into two files?

database.go to fetch all necessary data points and then dataprocessing.go to work with it? Then you can write a test for dataprocessing.go 

The test will be definitely necessary to ensure quality. Especially the splitting up of orders is in my experience very error prone and users do not tolerate any errors here ("if this is not correct, how can I trust the rest of the data?")

Reply:
Fixed in 11be320b0bee8e43fe0298450b5c2e8fed8dcdaa

Reply:
The test is still open. Can you write some with also some edge cases (e.g., multiple orders, etc.)?

Reply:
Yes, im currently working on that